src/uproot/models/RNTuple.py

```diff
@@ -480,10 +493,28 @@ def read_pagedesc(self, destination, desc, dtype_str, dtype, nbits, split):
         # needed to chop off extra bits incase we used `unpackbits`
         destination[:] = content[:num_elements]
 
-    def read_col_pages(self, ncol, cluster_range, pad_missing_ele=False):
-        res = numpy.concatenate(
-            [self.read_col_page(ncol, i) for i in cluster_range], axis=0
-        )
+    def read_col_pages(self, ncol, cluster_range, dtype_byte, pad_missing_ele=False):
+        arrays = [self.read_col_page(ncol, i) for i in cluster_range]
+
+        # Check if column stores offset values for jagged arrays (splitindex64) (applies to cardinality cols too):
+        if (
+            dtype_byte == uproot.const.rntuple_col_type_to_num_dict["splitindex64"]
+            or dtype_byte == uproot.const.rntuple_col_type_to_num_dict["splitindex32"]
+        ):
+            # Extract the last offset values:
+            last_elements = [
+                arr[-1] for arr in arrays[:-1]
+            ]  # First value always zero, therefore skip first arr.
+            # Compute cumulative sum using itertools.accumulate:
+            last_offsets = [*list(accumulate(last_elements))]
```

Comment:
Also, this can now be simplified

```suggestion
            last_offsets = list(accumulate(last_elements))
```